ibc-clients/cw-context/src/context/client_ctx.rs

```diff
@@ -195,3 +196,37 @@ where
         Ok(())
     }
 }
+
+pub trait CwClientValidation<'a>: ClientValidationContext {
+    fn cosmwasm_query_context(&self) -> Option<&Deps<'a>>;
+    fn cosmwasm_execute_context(&mut self) -> Option<&mut DepsMut<'a>>;
+    fn generate_sha256_digest(&self, data: &[u8]) -> Checksum;
```

Comment:
```suggestion
    fn deps(&self) -> Option<&Deps<'a>>;
    fn deps_mut(&mut self) -> Option<&mut DepsMut<'a>>;
    fn checksum(&self, data: &[u8]) -> Checksum;
```

Reply:
Done in ca28a50bf49637bf86ee4a0eeac7393e88cd343a